specifyweb/frontend/js_src/lib/components/Toolbar/TreeRepair.tsx

```diff
@@ -67,6 +66,67 @@ export function TreeSelectDialog({
   const [treeRanks] = usePromise(treeRanksPromise, true);
   const [isFinished, setIsFinished] = useBooleanState();
 
+  const treeList = React.useMemo(
```

Comment:
this useMemo is useless right now because a new `getLint` function is passed on each render
could you please refactor to only have this part in the useMemo as this part is less likely to change often:

```ts
      getDisciplineTrees()
        .filter((treeName) =>
          permissionName === 'repair'
            ? hasPermission(`/tree/edit/${toLowerCase(treeName)}`, 'repair')
            : hasTreeAccess(treeName, 'read')
        )
        .map((treeName) => {
          const treeDefinition = deserializeResource(
            getTreeDefinitions(treeName)[0].definition
          );
          return [treeName, treeDefinition] as const;
      }
```

and outside of the useMemo you would map over the results to render the actual elements